ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java

```diff
@@ -932,9 +896,20 @@ private boolean isStringInUrl(String toFind, int timeoutSeconds) {
         return result;
     }
 
-    @When("^set 3scale discovery variable to \"([^\"]*)\"")
+    @When("^enable 3scale discovery with url \"([^\"]*)\"")
     public void set3scaleEnvVar(String value) {
-        OpenShiftUtils.updateEnvVarInDeploymentConfig("syndesis-server", "CONTROLLERS_EXPOSE_VIA3SCALE", value);
+        String feature = "\"managementUrlFor3scale\":\"" + value + "\"";
+        if ("".equals(value)) {
+            feature = "";
+        }
+        HTTPResponse rsp = OpenShiftUtils.invokeApi(
+            HttpUtils.Method.PATCH,
```

Comment:
You can now use methods in the `SyndesisTemplate` class to edit the CR:
https://github.com/syndesisio/syndesis-qe/blob/master/utilities/src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java#L90-L96

The `getDeployedCr` will return the CR as a hierarchy of Maps, you can modify that and pass it `editCr`. Working with the maps is not super convenient (lots of type casting and `computeIfAbsent`), but refactoring that into something more pleasant is on my radar.

Reply:
ok thanks, will give it a look and redo it :)